src/rgw/rgw_rados.cc

```diff
@@ -12257,23 +12257,42 @@ int RGWRados::remove_objs_from_index(RGWBucketInfo& bucket_info, list<rgw_obj_in
   int r = open_bucket_index(bucket_info, index_ctx, dir_oid);
   if (r < 0)
     return r;
-
-  bufferlist updates;
-
-  for (auto iter = oid_list.begin(); iter != oid_list.end(); ++iter) {
-    rgw_bucket_dir_entry entry;
-    entry.key = *iter;
-    dout(2) << "RGWRados::remove_objs_from_index bucket=" << bucket_info.bucket << " obj=" << entry.key.name << ":" << entry.key.instance << dendl;
-    entry.ver.epoch = (uint64_t)-1; // ULLONG_MAX, needed to that objclass doesn't skip out request
-    updates.append(CEPH_RGW_REMOVE | suggest_flag);
-    ::encode(entry, updates);
+  {
```

Comment:
fix bug : in the case,  we remove a lot of objs at a time, we will get a errno, -90 (mes is too long), will remove 1000 files per time